### PR TITLE
Update CMake

### PR DESCRIPTION
**Pull request by tmmsartor**
Add the possibility to compile HPIPM through CMake build system from a generic third-pary
library.
- Overridable BLASFEO path or library  
- Overridable HPIPM installation paths
- Avoid hard override of other build variables

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,33 +32,39 @@ project(hpipm)
...
-	set(BLASFEO_PATH /opt/blasfeo)
+# Target architecture
+set(TARGET GENERIC CACHE STRING "Set CPU architecture target")
+
+# Installation
+set(CMAKE_INSTALL_PREFIX "/opt/hpipm"  CACHE STRING "Installation path")
+
+# BLASFEO Option
+set(BLASFEO_PATH /opt/blasfeo CACHE STRING "BLASFEO installation path")
+set(HPIPM_BLASFEO_LIB "Static" CACHE STRING "BLASFEO installation path")
```
Comment (giaf) [inline CMakeLists.txt]: `STRING` should not be `BLASFEO installation path` I guess

File: CMakeLists.txt [hunk 2]
```diff
@@ -32,33 +32,39 @@ project(hpipm)
...
-#set(REF_BLAS NETLIB)
-#set(REF_BLAS MKL)
-#set(REF_BLAS BLIS)
-#set(REF_BLAS ATLAS)
+set(REF_BLAS 0 CACHE STRING "Set CPU architecture target")
+set(REF_BLAS OPENBLAS CACHE STRING "Set CPU architecture target")
+set(REF_BLAS NETLIB   CACHE STRING "Set CPU architecture target")
+set(REF_BLAS MKL      CACHE STRING "Set CPU architecture target")
+set(REF_BLAS BLIS     CACHE STRING "Set CPU architecture target")
+set(REF_BLAS ATLAS    CACHE STRING "Set CPU architecture target")
```
Comment (giaf) [inline CMakeLists.txt]: Which one is set here? `ATLAS`?

File: CMakeLists.txt [hunk 3]
```diff
@@ -32,33 +32,39 @@ project(hpipm)
...
 
-# BLASFEO
-if(NOT TARGET blasfeo)
-	# BLASFEO installation directory
-	set(BLASFEO_PATH /opt/blasfeo)
+# Target architecture
+set(TARGET GENERIC CACHE STRING "Set CPU architecture target")
+
+# Installation
+set(CMAKE_INSTALL_PREFIX "/opt/hpipm"  CACHE STRING "Installation path")
```
Comment (giaf) [inline CMakeLists.txt]: is it correct the path as a string (i.e. with ".."), while `BLASFEO_PATH` has not?

File: CMakeLists.txt [hunk 4]
```diff
@@ -32,33 +32,39 @@ project(hpipm)
...
-# Options
 # set(HPIPM_TESTING ON CACHE BOOL "Tests enabled")
 set(HPIPM_TESTING OFF CACHE BOOL "Tests disabled")
 
-# BLASFEO
-if(NOT TARGET blasfeo)
-	# BLASFEO installation directory
-	set(BLASFEO_PATH /opt/blasfeo)
+# Target architecture
+set(TARGET GENERIC CACHE STRING "Set CPU architecture target")
```
Comment (giaf) [inline CMakeLists.txt]: I guess we can put `HASWELL` as default by now

Comment (giaf) [reply CMakeLists.txt]: Also in `Makefile.rule` is `AVX`

File: CMakeLists.txt (hunk 1 above)
Comment (tmmsartor) [reply CMakeLists.txt]: True, thanks!

File: CMakeLists.txt (hunk 4 above)
Comment (tmmsartor) [reply CMakeLists.txt]: Fixed in latest commits.

File: CMakeLists.txt (hunk 3 above)
Comment (tmmsartor) [reply CMakeLists.txt]: In this case should be the same, in general there are some differences, i.e. with list
https://stackoverflow.com/questions/35847655/when-should-i-quote-variables
I quoted everything in the latest commit.

File: CMakeLists.txt (hunk 2 above)
Comment (tmmsartor) [reply CMakeLists.txt]: True, thanks, fixed.

**Pull request merged**